client/src/components/GreensockComponent/GreensockComponent.js

```diff
@@ -0,0 +1,12 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+const GreensockComponent = ({ message }) => {
```

Comment:
Let's remove this Greensock component for now. It's not required IMO. We will be having component that will use Greensock and not a Greensock component itself.

Reply:
So, the current setup is, 
Route: `home`
      Page: `home`
          Containers: `Header` and `Footer` and `Greensock`
Each container defines the render properties, default or otherwise
Each container then, passes these properties to its associated component.

The component associated with the container, renders the content.

So we need three components, for each of the three containers. 
  - Containers prep data,
  - Accompanying Components wrap data inside markup and render.

That was how, the sample container was used by the page home which rendered "This is a sample page" on screen.

I am unsure of how greensock itself works At the moment. I am assuming, 

 - We get an animatable object which in turn has 
    - the quote
    - the question
    - the list of answers

these need a container to fit into. 
the greensock container, will prep the data as required, and make them renderable properties
the greensock container then binds this renderable property to it's component.
the greensock component defines its animations
the greensock component then renders animated content.

the current greensock component, is only a placeholder, and is not coded. Not even a single line of js. 

maybe we can keep it that way, so when we need to write the associated JS, we have a placeholder component to edit?

Reply:
GreenSock has very simple syntax such as 
TweenMax(#id, duration, {css property object}) that's it and the `div` with `id` gets animated. This can be done in `componnetDidMount` method of the component. IMO separate greensock container is not required.

Reply:
Are you saying that GreenSock container will wrap every animatable object?

Reply:
I think we should use general Content container and add components inside it. The components can use greensock api to provide animations.

Reply:
To be clear, What I am saying, this: 

1. We need 4 Sections (Header+Footer+Content+Nav) (*Everybody agrees?*) (shaan has reservations for the footer)

2. Now, If we have a footer, we can attribute our sources there, and have our link backs to the repo

3. I do not know yet, what forms the header, we can decide that later. We now just need a Empty section for the header.

4. Between the header and the footer Sits the **content** . This Section, will have 

    a. The Quote, 
    b. The Answers
    c. Anything else necessary

5. If the above is agreed upon, Then, **Where** are we getting the content from ? From a data API(either our won / third party)

6. Where and How do we display these quotes and their paraphernalia ?

     a. We prepare the **Render Props** in the **Content Container** , and then pass them to the **Content Component**
     b. If as @twishasaraiya  mentions, Greensock is easy to setup, It needs a DOM ID. 
     c. Keeping (b) in mind, this DOM element, Must have rendered our content from (5)

And so, we need the content container, bound to  the content component
All with me? I am coming to why I chose to do it that way in a bit. 

### Now, How Is the **Initial** **React generated code** laid out? 

1. There is a `PAGE`

2. A `PAGE` has a `CONTAINER` (`sampleContainer`)

3. The SampleContainer , **defines** all **Properties** that represent the data(**Model**)

4. The sampleContainer then, **Binds** to a **Component** the `sampleComponent` and passes renderable props. 

**The above** is how, react does it. 

### What we are doing in this PR ?

1. We are **copying** how react itself generates an app. 

2. We are copying react's `sampleContainer` to three copies (Header/Footer/Content). We can always add Nav. 

3. We are then copying react's `sampleComponent` to three copies (Header/Footer/Content). 

And we are changing **names**. That is all :) 

We are not **coding** a component. 

I guess, all that coding can be done later. **Incrementally** in an **Agile** manner, By **Multiple** contributors. 

The source of all confusion, rests on the fact that, I have named the **content** container /component in the list above, as **greensock**. 

I can always change it to **content**, so there is no confusion.

Now, This PR, is only about, 

1. Defining three `sections`, the react way (as in sampleContainer/sampleComponent and how they are bound and laid out on page) BY:
a. Defining a `react container`  and `react component` for these three sections that we need
b. Defining a `react container to react component` binding, **so we retain**  a `react` way of logically seperating concerns 

4. We are then finally, defining the `app` as a `flexbox`, and making the sections responsive.


###What we are not doing

1. Deciding how we animate
2. Deciding how data is stored
3. Deciding / Building the intermediate / final UI 

Which can be done either by the lot of us discussing this PR, or by others, at their own times. 
I am open to finishing the complete UI too, but this, is just to define the emptyness and organize it for now.

Reply:
Yes, @retrocausal it is clear to me what you are trying to do. But `ContentContainer` doesn't sound reasonable to me. We can have a `ContentComponent` though which wraps everything inside it into `<section>` tag with some default styles for content. But mostly we would have a `QuizContainer` which will contain `QuestionComponent` and different `OptionComponent`.

It's not necessary to always have a container for every single component. But a component should only be included in a container. Like above quiz is handling question and options.

Let's do this:
1. Keep the Header (Can be used to show the hamburger menu and the score and other things)
2. Create a Sidebar, which will slide-in when we tap on hamburger menu.
3. Let's have a Navigation inside the Sidebar.
4. Keep the Footer and we will see if we need it. If there is just static data and we don't really need any state in this, we can skip FooterContainer and just keep the component.

This summarises what the app can actually use. Get in touch with me if you have any doubts or queries :)

Reply:
That clears things up! will make these changes, and commit. I just wanted to have one exhaustive comment, and clarify just in case I was infusing confusion in to the unusually long thread on this PR :laughing: 
Thanks Shaan!